### PR TITLE
Gather all types in a single place

### PR DESCRIPTION
**Pull request by AdrienCorenflos**

### CODE REVIEW

File: blackjax/common.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from typing import Dict, List, Tuple, Union
```
Comment (rlouf) [inline blackjax/common.py:1]: Could we name it `types.py` ?

**Comment by rlouf** [comment]
Thank you for opening this PR. Now we are working on meta-algorithms (SMC, parallel tempering, coupling) it is time that we start thinking about types across the library.

File: blackjax/common.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+import jax.numpy as jnp
+
+Array = Union[jnp.ndarray, jnp.DeviceArray, float]
```
Comment (junpenglao) [inline blackjax/common.py:5]:
```suggestion
Array = Union[np.ndarray, jnp.DeviceArray]
```

**Review by rlouf — CHANGES_REQUESTED** [review]
I think we can merge once the `PRNGKey` type is added.

**Comment by AdrienCorenflos** [comment]
You sure you want to do this? It feels like it may be confusing for end
users...

**Comment by rlouf** [comment]
How would it be confusing ? Because the name of the function that returns a key is `jax.random.PRNGKey`?

It may be for a split second for someone who edits the code, but I think would be very readable for others.

**Comment by rlouf** [comment]
Interested in knowing why you thought this would be confusing.

**Comment by rlouf** [comment]
Looks like you need to import the names in many files still, and fix the linting. After that we're good to merge.

**Comment by junpenglao** [comment]
Nice! Thanks @AdrienCorenflos!

**Pull request merged**